app/src/renderer/views/Inbox/MainContent/Conversation/Item/File.tsx

```diff
@@ -14,21 +14,176 @@ import "../Item.css";
 import "./File.css";
 
 import { useTranslation } from "react-i18next";
-import {
-  File as FileIcon,
-  Download,
-  LoaderCircle,
-  FileCheck2,
-  FileX2,
-} from "lucide-react";
+import { memo } from "react";
+import { File as FileIcon, Download, LoaderCircle, FileX2 } from "lucide-react";
 import { Button, Tooltip } from "antd";
+import {
+  FilePdfFilled,
+  FileExcelFilled,
+  FileImageFilled,
+  FileWordFilled,
+  FileZipFilled,
+  FilePptFilled,
+  FileMarkdownFilled,
+  FileFilled,
+} from "@ant-design/icons";
+import FileVideoFilled from "./FileVideoFilled";
+import FileAudioFilled from "./FileAudioFilled";
+
+const MemoizedFileVideoFilled = memo(FileVideoFilled);
+const MemoizedFileAudioFilled = memo(FileAudioFilled);
+
+const EXCEL_EXTENSIONS = new Set([
+  ".xls",
+  ".xlsx",
+  ".xlsm",
+  ".xlsb",
+  ".xlt",
+  ".xltx",
+  ".xltm",
+  ".csv",
+  ".tsv",
+  ".ods",
+]);
+
+const IMAGE_EXTENSIONS = new Set([
+  ".png",
+  ".jpg",
+  ".jpeg",
+  ".gif",
+  ".bmp",
+  ".tif",
+  ".tiff",
+  ".svg",
+  ".svgz",
+  ".webp",
+  ".ico",
+  ".djvu",
+  ".heif",
+  ".heic",
+  ".avif",
+]);
+
+const WORD_EXTENSIONS = new Set([".doc", ".docx", ".rtf", ".odt", ".txt"]);
+
+const ARCHIVE_EXTENSIONS = new Set([
+  ".zip",
+  ".gz",
+  ".tgz",
+  ".tar",
+  ".bz2",
+  ".tbz",
+  ".xz",
+  ".txz",
+  ".7z",
+  ".rar",
+]);
+
+const POWERPOINT_EXTENSIONS = new Set([
+  ".ppt",
+  ".pptx",
+  ".pps",
+  ".ppsx",
+  ".odp",
+]);
+
+const VIDEO_EXTENSIONS = new Set([
+  ".mp4",
+  ".mov",
+  ".avi",
+  ".mkv",
+  ".wmv",
+  ".flv",
+  ".webm",
+  ".ogv",
+  ".m4v",
+  ".mpg",
+  ".mpeg",
+  ".3gp",
+  ".3g2",
+]);
+
+const AUDIO_EXTENSIONS = new Set([
+  ".mp3",
+  ".wav",
+  ".flac",
+  ".aac",
+  ".ogg",
+  ".oga",
+  ".m4a",
+  ".wma",
+  ".aif",
+  ".aiff",
+  ".opus",
+]);
 
 interface FileProps {
   item: Item;
   designation: string;
   onUpdate: (update: ItemUpdate) => void;
 }
 
+function getFileIconAndColor(filename: string): {
+  Icon: React.ComponentType<{
+    style?: React.CSSProperties;
+    className?: string;
+  }>;
+  color: string;
+} {
+  const normalizedFilename = filename.toLowerCase();
+  const lastDotIndex = normalizedFilename.lastIndexOf(".");
```

Comment:
nit: You could use https://nodejs.org/api/path.html#pathextnamepath which is basically the same thing.